packages/arb-token-bridge-ui/src/util/deposits/helpers.ts

```diff
@@ -265,6 +303,168 @@ const updateClassicDepositStatusData = async ({
   return completeDepositTx
 }
 
+async function getTimestampResolved(
+  destinationChainProvider: Provider,
+  l3TxHash: string | undefined
+) {
+  if (typeof l3TxHash === 'undefined') {
+    return
+  }
+  return await destinationChainProvider
+    .getTransactionReceipt(l3TxHash)
+    .then(tx => tx.blockNumber)
+    .then(blockNumber => destinationChainProvider.getBlock(blockNumber))
+    .then(block => String(block.timestamp * 1000))
+}
+
+export async function fetchTeleporterDepositStatusData({
+  assetType,
+  sourceChainId,
+  destinationChainId,
+  txId
+}: {
+  assetType: AssetType
+  sourceChainId: number
+  destinationChainId: number
+  txId: string
+}): Promise<{
+  status?: TxnStatus
+  timestampResolved?: string
+  l1ToL2MsgData?: L1ToL2MessageData
+  l2ToL3MsgData?: L2ToL3MessageData
+}> {
+  const isNativeCurrencyTransfer = assetType === AssetType.ETH
+  const sourceChainProvider = getProviderForChainId(sourceChainId)
+  const destinationChainProvider = getProviderForChainId(destinationChainId)
+  const { l2ChainId } = await getL2ConfigForTeleport({
+    destinationChainProvider
+  })
+
+  function isEthTeleport(
+    status: EthTeleportStatus | Erc20TeleportStatus
+  ): status is EthTeleportStatus {
+    return isNativeCurrencyTransfer
+  }
+
+  try {
+    const depositStatus = await fetchTeleportStatusFromTxId({
+      txId,
+      sourceChainProvider,
+      destinationChainProvider,
+      isNativeCurrencyTransfer
+    })
+    const l2ToL3MsgData: L2ToL3MessageData = {
+      status: L1ToL2MessageStatus.NOT_YET_CREATED,
+      l2ChainId
+    }
+    const l2Retryable = isEthTeleport(depositStatus)
+      ? depositStatus.l2Retryable
+      : depositStatus.l1l2TokenBridgeRetryable
+    const l2ForwarderFactoryRetryable = isEthTeleport(depositStatus)
+      ? null
+      : depositStatus.l2ForwarderFactoryRetryable
+    const l3Retryable = isEthTeleport(depositStatus)
+      ? depositStatus.l3Retryable
+      : depositStatus.l2l3TokenBridgeRetryable
+
+    // extract the l2 transaction details, if any
+    const l1l2Redeem = await l2Retryable.getSuccessfulRedeem()
+    const l1ToL2MsgData: L1ToL2MessageData = {
+      status: await l2Retryable.status(),
+      l2TxID:
+        l1l2Redeem && l1l2Redeem.status === L1ToL2MessageStatus.REDEEMED
+          ? l1l2Redeem.l2TxReceipt.transactionHash
+          : undefined,
+      fetchingUpdate: false,
+      retryableCreationTxID: l2Retryable.retryableCreationId
+    }
+
+    // in case the forwarder retryable has failed, add it to the `l2ToL3MsgData`, else leave it undefined
+    // note: having `l2ForwarderRetryableTxID` in the `l2ToL3MsgData` will mean that it needs redemption
+    if (
+      !depositStatus.completed &&
+      l2ForwarderFactoryRetryable &&
+      (await l2ForwarderFactoryRetryable.status()) ===
```

Comment:
since we a re using `await l2ForwarderFactoryRetryable.status()` below at the return object, we should abstract it to a variable